clients/src/main/java/org/apache/kafka/clients/consumer/NoOffsetForPartitionException.java

```diff
@@ -26,19 +26,20 @@
 
     private static final long serialVersionUID = 1L;
 
-    private final TopicPartition partition;
+    private final Set<TopicPartition> partitionSet;
 
-    public NoOffsetForPartitionException(TopicPartition partition) {
-        super("Undefined offset with no reset policy for partition: " + partition);
-        this.partition = partition;
+    public NoOffsetForPartitionException(String message) {
```

Comment:
@hachikuji I agree that evaluating these options would be valuable and the error handling has room for improvement. I also think this patch, that removes the duplicate code/functionality and cleans up some missing exceptions in the Errors class, is a step in that direction. 

Do you think that effort should block this patch?